internal/clients/etherscan/pubkey-finder_test.go

```diff
@@ -0,0 +1,127 @@
+package etherscan
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"testing"
+)
+
+func TestGetFromResultHash(t *testing.T) {
+	mockTxResult := []txResult{
+		{From: "address1", Hash: "aaa111"},
+		{From: "address2", Hash: "bbb222"},
+		{From: "address3", Hash: "ccc333"},
+	}
+
+	type input struct {
+		address string
+		txList  *txList
+	}
+	type output struct {
+		hash         common.Hash
+		wantsErr     bool
+		errorMessage string
+	}
+	testCases := []struct {
+		name   string
+		input  input
+		output output
+	}{
+		{
+			"empty-transaction-list",
+			input{
+				"should-not-matter",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  []txResult{},
+				},
+			},
+			output{
+				common.Hash{},
+				true,
+				fmt.Sprintf("No transactions found for address: %v", "should-not-matter"),
+			},
+		},
+		{
+			"match-transaction-1",
+			input{
+				"address1",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			output{
+				common.HexToHash("aaa111"),
+				false,
+				"",
+			},
+		},
+		{
+			"match-transaction-2",
+			input{
+				"address2",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			output{
+				common.HexToHash("bbb222"),
+				false,
+				"",
+			},
+		},
+		{
+			"match-transaction-3",
+			input{
+				"address3",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			output{
+				common.HexToHash("ccc333"),
+				false,
+				"",
+			},
+		},
+		{
+			"no-matching-transactions",
+			input{
+				"address11",
+				&txList{
+					Status:  "",
+					Message: "",
+					Result:  mockTxResult,
+				},
+			},
+			output{
+				common.Hash{},
+				true,
+				"No transactions from address found",
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			hash, err := getFromResultHash(testCase.input.address, testCase.input.txList)
+
+			if testCase.output.wantsErr {
+				if hash != testCase.output.hash || err == nil || err.Error() != testCase.output.errorMessage {
+					t.Errorf("getFromResultHash => Expected hash: %v and error containing message: %v. Got hash: %v and error: %v", testCase.output.hash, testCase.output.errorMessage, hash, err)
+				}
+			} else {
+				if hash != testCase.output.hash || err != nil {
+					t.Errorf("getFromResultHash => Expected hash: %v and error nil. Got hash: %v and error: %v", testCase.output.hash, hash, err)
+				}
+			}
+		})
+	}
+}
```

Comment:
This is quite difficult to read. Try looking at `TestPrivateKeyFromBytes` in `crypto/secp256k1/private_test.go` for a simple example of testing return values.